### PR TITLE
Add basic Jest setup

### LINKED ISSUE #39: [Testing] Set up Jest
Set up Jest for testing the React components

### PR DESCRIPTION
**Pull request by bkjohnson**
## Closes #39 

To run the tests, from the `rails` directory run:

```
npm run jest
```

### CODE REVIEW

File: rails/spec/javascript/StoryList.test.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import React from 'react'
+import StoryList from '../../app/javascript/components/StoryList'
+import { render } from 'react-testing-library'
+
+import { stories } from './fakes/stories'
+
+describe('<StoryList> component', () => {
+  it('renders correctly with a list of stories', () => {
+
+    const {container} = render( <StoryList stories={stories}/>)
```
Comment (baweaver) [inline rails/spec/javascript/StoryList.test.jsx]: Super nit: extra blank line

File: rails/spec/javascript/__snapshots__/Filter.test.jsx.snap [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (baweaver) [inline rails/spec/javascript/__snapshots__/Filter.test.jsx.snap]: Do we want to have the snapshots in here?

**Comment by cmhnk** [comment]
@bkjohnson @rudokemper We should add these instructions to the `setup.md` file and/or create a new testing instructions doc!

**Comment by bkjohnson** [comment]
That's a good point.  Speaking of that, what about using Jenkins (or something similar) to ensure that only code that passes tests can get merged?

**Comment by baweaver** [comment]
We'd mentioned using CircleCI in channel

**Pull request merged**